src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypeLikelihoodCalculator.java

```diff
@@ -0,0 +1,620 @@
+     *     {@link #ensureReadCapacity(int) ensureReadCapacity}.
+     * </p>
+     */
+    private final double[][] readLikelihoodsByGenotypeIndex;
+
+    /**
+     * Indicates how many reads the calculator supports.
+     *
+     * <p>This figure is increased dynamically as per the
+     * calculation request calling {@link #ensureReadCapacity(int) ensureReadCapacity}.<p/>
+     */
+    private int readCapacity = -1;
+
+    /**
+     * Caches the log10 of the first few integers up to the ploidy supported by the calculator.
+     * <p>This is in fact a shallow copy if {@link GenotypeLikelihoodCalculators#ploidyLog10}</p> and is not meant to be modified by
+     * this class. </p>
+     */
+    private final double[] log10;
+
+    /**
+     * Buffer field use as a temporal container for sorted allele counts when calculating the likelihood of a
+     * read in a genotype.
+     * <p>
+     *      This array follows the same format as {@link GenotypeAlleleCounts#sortedAlleleCounts}. Each component in the
+     *      genotype takes up two positions in the array where the first indicate the allele index and the second its frequency in the
+     *      genotype. Only non-zero frequency alleles are represented, taking up the first positions of the array.
+     * </p>
+     *
+     * <p>
+     *     This array is sized so that it can accommodate the maximum possible number of distinct alleles in any
+     *     genotype supported by the calculator, value stored in {@link #maximumDistinctAllelesInGenotype}.
+     * </p>
+     */
+    private final int[] genotypeAllelesAndCounts;
+
+    /**
+     * Buffer field use as a temporal container for component likelihoods when calculating the likelihood of a
+     * read in a genotype. It is stratified by read and the allele component of the genotype likelihood... that is
+     * the part of the likelihood sum that correspond to a particular allele in the genotype.
+     *
+     * <p>
+     *     It is implemented in a 1-dimensional array since typically one of the dimensions is rather small. Its size
+     *     is equal to {@link #readCapacity} times {@link #maximumDistinctAllelesInGenotype}.
+     * </p>
+     *
+     * <p>
+     *     More concretely [r][i] == log10Lk(read[r] | allele[i]) + log(freq[i]) where allele[i] is the ith allele
+     *     in the genotype of interest and freq[i] is the number of times it occurs in that genotype.
+     * </p>
+     */
+    private double[] readGenotypeLikelihoodComponents;
+
+    /**
+     * Creates a new calculator providing its ploidy and number of genotyping alleles.
+     */
+    protected GenotypeLikelihoodCalculator(final int ploidy, final int alleleCount,
+                                           final int[][] alleleFirstGenotypeOffsetByPloidy,
+                                           final GenotypeAlleleCounts[][] genotypeTableByPloidy,
+                                           final double[] ploidyLog10) {
```

Comment:
Same comment about the `LogCache`.  There's code in several places where a `ploidyLog10` array is calculated in order to be passed to this method, so make sure to remove those calculations.